Add results around Fig1C-D (#246)
AlexsLemonade/OpenPBTA-manuscript: content/03.results.md

We leveraged Docker® and the Rocker project [@arxiv:1710.03675] to create a monolithic image that contained all dependencies necessary for analyses in the analysis repository. 
To ensure that the image would build and that code would execute over time, we used the continuous integration (CI) service CircleCI® to run analytical code on a small subset of data for testing, which allowed us to detect any sensitivity to changes in the underlying data prior to code review (**Figure {@fig:Fig1}D**).

We followed a similar process in our Manubot-powered [@doi:10.1371/journal.pcbi.1007128] manuscript repository for additions to the manuscript (**Figure {@fig:Fig1}C**). 
Contributors forked the manuscript repository, added proposed content to their branch, and filed pull requests to the main manuscript repository with their changes.
Similarly, pull requests underwent a peer review process for clarity and correctness, agreement with interpretation, and spell checking via Manubot (**Figure {@fig:Fig1}C**).

<!-- TODO: Stub for the process for contributing analytical code (**Figure {@fig:S1}** and **STAR Methods**) and to the manuscript. -->

> ![**Overview of the OpenPBTA Project.** A, The Children's Brain Tumor Network and the Pacific Pediatric Neuro-Oncology Consortium collected tumor samples from 943 patients. To date, 38 cell lines were created from tumor tissue and over 2000 specimens were sequenced (N = 1035 RNA-Seq, N = 940 WGS, and N = 32 WXS or Targeted panel). Data was harmonized by the Kids First Data Resource Center using an Amazon S3 framework within CAVATICA. B, Stacked bar plot summary of the number of biospecimens per phase of therapy per broad histology (GNG = ganglioglioma, LGG = low-grade glioma, DIPG = diffuse intrinsic pontine glioma, DMG = diffuse midline glioma, EPN = ependymoma, HGG = high-grade glioma, ATRT = atypical teratoid rhabdoid tumor, CNS-ET = CNS embryonal tumor, MB = medulloblastoma, PNF = plexiform neurofibroma, DNET = dysembryoplastic neuroepithelial tumor, CRANIO = craniopharyngioma, EWS = Ewing sarcoma, CPP = choroid plexus papilloma. C, Overview of the open analysis and manuscript contribution model. In the analysis GitHub repository, a contributor would propose an analysis that can be commented on by other participants. Contributors would then implement the analysis and file a request to add their changes to the analysis repository ("pull request"). Pull requests underwent review for scientific rigor and correctness of implementation. Pull requests were additionally checked to ensure that all software dependencies were included and the code was not sensitive to underlying data changes using container and continuous integration technologies. Finally, a contributor would file a pull request to the Manubot-powered manuscript repository that underwent review. D, A potential path for an analytical pull request. Arrows indicate revisions to a pull request. Prior to review, a pull request was tested for dependency installation and whether or not the code would execute. Pull requests also required approval by organizers and/or other contributors, who checked for scientific correctness. Panel A created with BioRender.com.](https://raw.githubusercontent.com/AlexsLemonade/OpenPBTA-analysis/489d729cde4318654a26b0a9ef0dee6d706352cd/figures/pngs/figure1.png?sanitize=true){#fig:Fig1 width="7in"}

Reviewer: ```suggestion
![**Overview of the OpenPBTA Project.** A, The Children's Brain Tumor Network and the Pacific Pediatric Neuro-Oncology Consortium collected tumor samples from 943 patients. To date, 38 cell lines were created from tumor tissue, and over 2000 specimens were sequenced (N = 1035 RNA-Seq, N = 940 WGS, and N = 32 WXS or Targeted panel). Data was harmonized by the Kids First Data Resource Center using an Amazon S3 framework within CAVATICA. B, Stacked bar plot summary of the number of biospecimens per phase of therapy per broad histology (GNG = ganglioglioma, LGG = low-grade glioma, DIPG = diffuse intrinsic pontine glioma, DMG = diffuse midline glioma, EPN = ependymoma, HGG = high-grade glioma, ATRT = atypical teratoid rhabdoid tumor, CNS-ET = CNS embryonal tumor, MB = medulloblastoma, PNF = plexiform neurofibroma, DNET = dysembryoplastic neuroepithelial tumor, CRANIO = craniopharyngioma, EWS = Ewing sarcoma, CPP = choroid plexus papilloma). C, Overview of the open analysis and manuscript contribution model. In the analysis GitHub repository, a contributor would propose an analysis that other participants can comment on. Contributors would then implement the analysis and file a request to add their changes to the analysis repository ("pull request"). Pull requests underwent review for scientific rigor and correctness of implementation. Pull requests were additionally checked to ensure that all software dependencies were included and the code was not sensitive to underlying data changes using container and continuous integration technologies. Finally, a contributor would file a pull request to the Manubot-powered manuscript repository that underwent review. D, A potential path for an analytical pull request. Arrows indicate revisions to a pull request. Prior to review, a pull request was tested for dependency installation and whether or not the code would execute. Pull requests also required approval by organizers and/or other contributors, who checked for scientific correctness. Panel A created with BioRender.com.](https://raw.githubusercontent.com/AlexsLemonade/OpenPBTA-analysis/489d729cde4318654a26b0a9ef0dee6d706352cd/figures/pngs/figure1.png?sanitize=true){#fig:Fig1 width="7in"}
```

Something is wrong here? 
>Finally, a contributor would file a pull request to the Manubot-powered manuscript repository that underwent review.

The repository does not undergo review. Do you mean "Finally, a contributor would file a pull request to the Manubot-powered manuscript repository _to document their analysis_ that underwent review."?

Author: The pull request they file to the manuscript repository _does_ undergo review (that's what we're doing now!). Can rephrase.

> ![**Overview of the OpenPBTA Project.** A, The Children's Brain Tumor Network and the Pacific Pediatric Neuro-Oncology Consortium collected tumor samples from 943 patients. To date, 38 cell lines were created from tumor tissue, and over 2000 specimens were sequenced (N = 1035 RNA-Seq, N = 940 WGS, and N = 32 WXS or Targeted panel). Data was harmonized by the Kids First Data Resource Center using an Amazon S3 framework within CAVATICA. B, Stacked bar plot summary of the number of biospecimens per phase of therapy per broad histology (GNG = ganglioglioma, LGG = low-grade glioma, DIPG = diffuse intrinsic pontine glioma, DMG = diffuse midline glioma, EPN = ependymoma, HGG = high-grade glioma, ATRT = atypical teratoid rhabdoid tumor, CNS-ET = CNS embryonal tumor, MB = medulloblastoma, PNF = plexiform neurofibroma, DNET = dysembryoplastic neuroepithelial tumor, CRANIO = craniopharyngioma, EWS = Ewing sarcoma, CPP = choroid plexus papilloma). C, Overview of the open analysis and manuscript contribution model. In the analysis GitHub repository, a contributor would propose an analysis that other participants can comment on. Contributors would then implement the analysis and file a request to add their changes to the analysis repository ("pull request"). Pull requests underwent review for scientific rigor and correctness of implementation. Pull requests were additionally checked to ensure that all software dependencies were included and the code was not sensitive to underlying data changes using container and continuous integration technologies. Finally, a contributor would file a pull request to the Manubot-powered manuscript repository documenting their methods and results. Pull requests in the manuscript repository were also subject to review. D, A potential path for an analytical pull request. Arrows indicate revisions to a pull request. Prior to review, a pull request was tested for dependency installation and whether or not the code would execute. Pull requests also required approval by organizers and/or other contributors, who checked for scientific correctness. Panel A created with BioRender.com.](https://raw.githubusercontent.com/AlexsLemonade/OpenPBTA-analysis/489d729cde4318654a26b0a9ef0dee6d706352cd/figures/pngs/figure1.png?sanitize=true){#fig:Fig1 width="7in"}

Author: ```suggestion
![**Overview of the OpenPBTA Project.** A, The Children's Brain Tumor Network and the Pacific Pediatric Neuro-Oncology Consortium collected tumor samples from 943 patients. To date, 38 cell lines were created from tumor tissue, and over 2000 specimens were sequenced (N = 1035 RNA-Seq, N = 940 WGS, and N = 32 WXS or Targeted panel). Data was harmonized by the Kids First Data Resource Center using an Amazon S3 framework within CAVATICA. B, Stacked bar plot summary of the number of biospecimens per phase of therapy per broad histology (GNG = ganglioglioma, LGG = low-grade glioma, DIPG = diffuse intrinsic pontine glioma, DMG = diffuse midline glioma, EPN = ependymoma, HGG = high-grade glioma, ATRT = atypical teratoid rhabdoid tumor, CNS-ET = CNS embryonal tumor, MB = medulloblastoma, PNF = plexiform neurofibroma, DNET = dysembryoplastic neuroepithelial tumor, CRANIO = craniopharyngioma, EWS = Ewing sarcoma, CPP = choroid plexus papilloma). C, Overview of the open analysis and manuscript contribution model. In the analysis GitHub repository, a contributor would propose an analysis that other participants can comment on. Contributors would then implement the analysis and file a request to add their changes to the analysis repository ("pull request"). Pull requests underwent review for scientific rigor and correctness of implementation. Pull requests were additionally checked to ensure that all software dependencies were included and the code was not sensitive to underlying data changes using container and continuous integration technologies. Finally, a contributor would file a pull request documenting their methods and results to the Manubot-powered manuscript repository. Pull requests in the manuscript repository were also subject to review. D, A potential path for an analytical pull request. Arrows indicate revisions to a pull request. Prior to review, a pull request was tested for dependency installation and whether or not the code would execute. Pull requests also required approval by organizers and/or other contributors, who checked for scientific correctness. Panel A created with BioRender.com.](https://raw.githubusercontent.com/AlexsLemonade/OpenPBTA-analysis/489d729cde4318654a26b0a9ef0dee6d706352cd/figures/pngs/figure1.png?sanitize=true){#fig:Fig1 width="7in"}
```

We then harnessed the benchmarking efforts of the Kids First Data Resource Center to develop a robust and reproducible data analysis workflow within the CAVATICA platform to perform the following primary somatic analyses: variant calling of single nucleotide variants (SNVs), copy number variants (CNVs), structural variants (SVs), and fusions (**Figure S1** - red boxes and **STAR Methods**). 
We maintained a data release folder on Amazon S3 containing merged files for each analysis, downloadable from the GitHub repository or open access CAVATICA project.
As new analytical products that we expected to be used across multiple analyses were produced (e.g., tumor mutation burden calculations) or issues with the data were identified, new releases were made available in a versioned manner.

> A key feature of this project was the open contribution model used for both analyses (i.e., analytical code) and scientific manuscript writing. 

Author: I like the addition of the word `innovative`. I am less sold on the reference to things being coupled. I don't think that's specific to this project – I expect many Manubot papers use version controlled figures, etc.

```suggestion
A key innovative feature of this project was the open contribution model used for both analyses (i.e., analytical code) and scientific manuscript writing. 
```

> A key feature of this project was the open contribution model used for both analyses (i.e., analytical code) and scientific manuscript writing. 

Reviewer 2: Sure. I used "coupled" because I didn't think the open manubot writing itself isn't innovative (or maybe it still is since it hasn't been done _much_?).

We created a public Github analysis repository (https://github.com/AlexsLemonade/OpenPBTA-analysis) to hold all code associated with analyses downstream of the Kids First Data Resource Center workflows and a GitHub manuscript repository (https://github.com/AlexsLemonade/OpenPBTA-manuscript) with ManuBot [@url:https://manubot.org/; @doi:10.1371/journal.pcbi.1007128] integration to enable manuscript creation using Markdown within GitHub.

> Additions to the manuscript often started with proposing an analysis in the analysis repository.

Author: Fine with me!

> A potential contributor could propose an analysis by filing an issue on the GitHub analysis repository. 
> Organizers for the project or other contributors with expertise had the opportunity to comment on the proposal (**Figure {@fig:Fig1}C**). 
> The contributor then forked the analysis repository (or make a copy for the purpose of adding code changes) and added their proposed analysis to their fork. 
> They then filed a pull request on GitHub, where they requested to add their proposed changes to the main repository. 
> All pull requests to the analysis repository underwent peer review by organizers or other contributors. 
> Analytical code reviews conducted by peer reviewers checked for correctness of analyses (**Figure {@fig:Fig1}C-D**) and maintainability and readability of code and documentation. 

Reviewer 2: ```suggestion
First, a potential contributor proposed an analysis by filing an issue on the GitHub analysis repository. 
Next, organizers for the project, or other contributors with expertise, had the opportunity to comment on the proposal (**Figure {@fig:Fig1}C**). 
The contributor then forked the analysis repository (or make a copy for the purpose of adding code changes) and added their proposed analysis to their fork. 
They then filed a pull request on GitHub, where they requested to add their proposed changes to the main repository. 
All pull requests to the analysis repository underwent peer review by organizers and/or other contributors. 
Analytical code reviews conducted by peer reviewers checked for correctness of analyses (**Figure {@fig:Fig1}C-D**) and maintainability and readability of code and documentation. 
```

The nature of the project required additional steps beyond peer review of analytical code (**Figure {@fig:Fig1}D**).

> Because multiple contributors were working on the code base and months would sometimes pass between updates to specific analysis modules, it was important to maintain a consistent software development environment to ensure that code could be revised. 

Reviewer 2: ```suggestion
Since multiple contributors were working on the code base and months would sometimes pass between updates to specific analysis modules, it was important to maintain a consistent software development environment to ensure that code could be revised. 
```

> We leveraged Docker® and the Rocker project [@arxiv:1710.03675] to create a monolithic image that contained all the dependencies necessary for the analyses in the analysis repository. 

Reviewer 2: ```suggestion
We leveraged Docker® and the Rocker project [@arxiv:1710.03675] to create a monolithic image that contained all dependencies necessary for analyses in the analysis repository. 
```

> We used continuous integration (CI) to make sure that the image would build and that code would execute (**Figure {@fig:Fig1}D**). 
> We also needed ensure that code did not cease to function over time, as data releases may introduce changes the order or number of samples. 
> Using CI to run analytical code on a small subset of data used specifically for testing allowed us to check for any sensitivity to changes in the underlying data prior to code review (**Figure {@fig:Fig1}D**).

Reviewer 2: ```suggestion
To ensure that the image would build and that code would execute over time, we used continuous integration (CI) to run analytical code on a small subset of data for testing, which allowed us to detect any sensitivity to changes in the underlying data prior to code review (**Figure {@fig:Fig1}D**).
```

> The Manubot-powered [@url:https://manubot.org/; @doi:10.1371/journal.pcbi.1007128] manuscript repository allowed us to follow a similar process for additions to this manuscript (**Figure {@fig:Fig1}C**). 

Author: A little more active

```suggestion
We followed a similar process in our Manubot-powered [@doi:10.1371/journal.pcbi.1007128] manuscript repository for additions to the manuscript (**Figure {@fig:Fig1}C**). 
```

> The Manubot-powered [@url:https://manubot.org/; @doi:10.1371/journal.pcbi.1007128] manuscript repository allowed us to follow a similar process for additions to this manuscript (**Figure {@fig:Fig1}C**). 

Reviewer 2: 👍 I was trying to not start so many sentences with "we", but it is hard to keep it all active, first person.

> Contributors forked the manuscript repository and added proposed methods and results to their branch.
> Contributors then filed pull requests to the main manuscript repository with their changes.
> These pull requests underwent a review process that included peer review for clarity and correctness, agreement with interpretation, and spell checking via Manubot (**Figure {@fig:Fig1}C**).

Reviewer 2: ```suggestion
Contributors forked the manuscript repository, added proposed content to their branch, and filed pull requests to the main manuscript repository with their changes.
Similarly, pull requests underwent a peer review process for clarity and correctness, agreement with interpretation, and spell checking via Manubot (**Figure {@fig:Fig1}C**).
```

We previously performed whole genome sequencing (WGS), whole exome sequencing (WXS), and RNA sequencing (RNA-Seq) on matched tumor and normal tissues as well as selected cell lines from 943 patient tumors from the Pediatric Brain Tumor Atlas (PBTA) [@doi:10.1093/neuonc/noz192], consisting of samples from the Children’s Brain Tumor Network (CBTN) [@url:https://CBTN.org] and the PNOC003 DMG clinical trial [@doi:10.1002/ijc.32258] of the Pacific Pediatric Neuro-Oncology Symposium (PNOC) (**Figure {@fig:Fig1}A**). 
We then harnessed the benchmarking efforts of the Kids First Data Resource Center to develop a robust and reproducible data analysis workflow within the CAVATICA platform to perform the following primary somatic analyses: variant calling of single nucleotide variants (SNVs), copy number variants (CNVs), structural variants (SVs), and fusions (**Figure S1** - red boxes and **STAR Methods**). 
We maintained a data release folder on Amazon S3 containing merged files for each analysis, downloadable from the GitHub repository or open access CAVATICA project (See **Data and code availability** section).

> As new analytical products that we expected to be used across multiple analyses were produced (e.g., tumor mutation burden calculations) or issues with the data were identified, new releases were made available in a versioned manner.

Reviewer: ```suggestion
As new analytical products (e.g., tumor mutation burden calculations) that we expected to be used across multiple analyses were produced, or issues with the data were identified, new releases were made available in a versioned manner.
```

I find it a bit clearer with this parenthetical placement

A key innovative feature of this project was the open contribution model used for both analyses (i.e., analytical code) and scientific manuscript writing. 
We created a public Github analysis repository (https://github.com/AlexsLemonade/OpenPBTA-analysis) to hold all code associated with analyses downstream of the Kids First Data Resource Center workflows and a GitHub manuscript repository (https://github.com/AlexsLemonade/OpenPBTA-manuscript) with ManuBot [@doi:10.1371/journal.pcbi.1007128] integration to enable real-time manuscript creation using Markdown within GitHub.

Additions to the manuscript often started with proposing a specific analysis in the analysis repository (for example, immune cell deconvolution of bulk RNA-Seq data).

> First, a potential contributor proposed an analysis by filing an issue on the GitHub analysis repository. 

Author: I 100% started out this way, so I am happy to revert to that at a later date.

Next, organizers for the project, or other contributors with expertise, had the opportunity to comment on the proposal (**Figure {@fig:Fig1}C**). 
The contributor then forked the analysis repository (or make a copy for the purpose of adding code changes) and added their proposed analysis to their fork. 
They then filed a pull request on GitHub, where they requested to add their proposed changes to the main repository. 
All pull requests to the analysis repository underwent peer review by organizers and/or other contributors. 

> Analytical code reviews conducted by peer reviewers checked for correctness of analyses (**Figure {@fig:Fig1}C-D**) and maintainability and readability of code and documentation. 

Reviewer: ```suggestion
Analytical code reviews conducted by peer reviewers checked for analysis accuracy (**Figure {@fig:Fig1}C-D**) and maintainability as well as readability of code and documentation. 
```

Personal preference - when I write `(1 and 2) and (3 and 4)`, I like to use _not and_ between them. So: `(1 and 2) as well as `3 and 4)`.

The nature of the project required additional steps beyond peer review of analytical code (**Figure {@fig:Fig1}D**).
Since multiple contributors were working on the code base and months would sometimes pass between updates to specific analysis modules, it was important to maintain a consistent software development environment to ensure that code could be revised. 
We leveraged Docker® and the Rocker project [@arxiv:1710.03675] to create a monolithic image that contained all dependencies necessary for analyses in the analysis repository. 

> To ensure that the image would build and that code would execute over time, we used continuous integration (CI) to run analytical code on a small subset of data for testing, which allowed us to detect any sensitivity to changes in the underlying data prior to code review (**Figure {@fig:Fig1}D**).

Author: ```suggestion
To ensure that the image would build and that code would execute over time, we used the continuous integration (CI) service CircleCI® to run analytical code on a small subset of data for testing, which allowed us to detect any sensitivity to changes in the underlying data prior to code review (**Figure {@fig:Fig1}D**).
```
